packages/dev/core/src/Sprites/spriteRenderer.ts

```diff
@@ -478,9 +479,10 @@ export class SpriteRenderer {
         }
 
         // Positions
-        this._vertexData[arrayOffset] = sprite.position.x;
-        this._vertexData[arrayOffset + 1] = sprite.position.y;
-        this._vertexData[arrayOffset + 2] = sprite.position.z;
+        const floatingOriginOffset = this._scene?.floatingOriginOffset || Vector3.ZeroReadOnly;

```

Comment:
I think you should pass `this._scene?.floatingOriginOffset || Vector3.ZeroReadOnly` to the function, because **_appendSpriteVertex** is called 1 to 4 times per sprite, and we can have quite a large number of sprites in a scene.